microsoft/testsuites/core/gdb.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from __future__ import annotations
+
+import os
+import re
+from pathlib import Path
+
+from assertpy import assert_that
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CBLMariner
+from lisa.tools import Gcc, Gdb
+from lisa.util import get_matched_str
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers gdb functionality.
+    """,
+)
+class GDB(TestSuite):
+    # Hello World![Inferior 1 (process 1869) exited normally]
+    _gdb_output_pattern = re.compile(
+        r".*Hello World!\[Inferior .* \(process .*\) exited normally\].*"
+    )
+
+    @TestCaseMetadata(
+        description="""
+        This test case check gdb work well.
+        """,
+        priority=1,
+    )
+    def verify_gdb(self, node: Node) -> None:
+        if type(node.os) == CBLMariner:
+            node.os.install_packages(("binutils", "glibc-devel"))
+        self._copy_to_node(node, "hello.c")
+        node.tools[Gcc].compile(
+            f"{node.working_path}/hello.c", f"{node.working_path}/hello", "-g -ggdb"
+        )
+        output = node.tools[Gdb].debug(f"{node.working_path}/hello")
+        matched = get_matched_str(output, self._gdb_output_pattern)
+        assert_that(matched).described_as(
+            f"dbg result doesn't matched expected value, actual output is {output}"
+        ).is_not_empty()
+
+    def _copy_to_node(self, node: Node, filename: str) -> None:
+        file_path = Path(os.path.dirname(__file__)) / "TestScripts" / filename
```

Comment:
This can be a const at class level, because the file is submitted, and never changed.

Reply:
updated